### ISSUE TITLE
Reporting

### ISSUE THREAD

**Issue by emaxsaun** [yahoo/protractor-retry#30]
Thank you for such a great plugin. I have been able to get retry working both locally, as well as in Jenkins, and Jenkins correctly fails the builds now, this is working as I would expect it to. My last roadblock to fully use protractor-retry is to be able to get reporting to work properly. I currently use this reporting plugin - protractor-jasmine2-screenshot-reporter, https://www.npmjs.com/package/protractor-jasmine2-screenshot-reporter, and it has worked well, but now that I'm adding in retry using protractor-retry, I have run into some issues with the reports generated. The plugin is only generating the most recently failed spec, so if I have a large suite of tests, the report with the failures will only one one suite. The report gets over written every time. Are there other reporting plugins that you guys can recommend that would provide all reports (first run report, rerun reports, etc)? I am not married to this specific reporter, I just need to get a report generated for all failures, not just the most recent spec/suite. Thank you.

**Comment by dreuxl** [comment]
Nice to read that it helps your projects.

We currently support genuine mocha & jasmine fwks. I need to check what this custom reporter s doing and how it works. I just checked the npm and it s adding config in the same area than our plugin so most likely something is messing up somewhere between those 2 :(

**Comment by emaxsaun** [comment]
Yeah. I use Jasmine. I am open to using another reporter. I just need something with screenshots. Ah, that is probably what is happening :(, do you know of any other reporters that have worked well with retry? Thanks for the quick response btw :)

**Comment by dreuxl** [comment]
we are using it with sauce labs which is providing video recording which is ok for our needs, but that s not helping your case I guess.

**Comment by amrot17** [comment]
@emaxsaun good to hear that this is finally working with your project 

Im using a mocha , with a custom reporter that adding the `webdriver.remote.sessionid` - knowing that this is a unique identifier -  to the generated report file name 

and I've modified the retry code to pass the report file name in the protractor-retry logs , so once the retry will be triggered , the assigned/generated report files will be deleted 

hope this would help in your case

**Comment by emaxsaun** [comment]
Sorry for the delayed response @dreuxl @amrot17. 

@amrot17 What custom reporter are you using? I am using Jasmine, not Mocha, in my case. 

Is there a way I can test your modified retry code? Is there a new version I can download using NPM?

**Comment by yuxi0203** [comment]
Hi @emaxsaun, I had similar situation since I am using [protractor-jasmine2-screenshot-reporter](https://www.npmjs.com/package/protractor-jasmine2-screenshot-reporter) as well. Since that reporter library needs to return promise, in my code, I am doing something like 
```
afterLaunch: (code) => new Promise((resolve) => {
        screenshotReporter.afterLaunch(() => resolve(code));
}).then(() => retry.afterLaunch(2)),
```

My guess is retry library just cleans-up screenshot folders during retry. 

The solution works for me is to using create new path/folder, checkout my following code
```
const RETRY_NUMBER = process.argv.indexOf('--retry') > -1 ? process.argv[process.argv.indexOf('--retry') + 1] : 0;
const HtmlScreenshotReporter = require('protractor-jasmine2-screenshot-reporter');
const screenshotReporter = new HtmlScreenshotReporter({ dest: `${documentationDir}/${RETRY_NUMBER}` });
``` 
And you can add path into `jasmine.getEnv().addReporter` as well. Let me know if that can solve your problem.

**Comment by emaxsaun** [comment]
@yuxi0203 Thanks for this, I am going to try to do this in my project. I will let you know if this fixes it for me.

**Comment by dreuxl** [comment]
Sweet to see that u guys can help each other :)

I'll add this `protractor-jasmine2-screenshot-reporter` reporter example, in our list of examples supported.

thanks again

**Comment by emaxsaun** [comment]
:) Np, thanks for adding that.
